pinot-broker/src/main/java/org/apache/pinot/broker/broker/helix/BaseBrokerStarter.java

```diff
@@ -346,9 +352,38 @@ public void start()
       timeSeriesRequestHandler = new TimeSeriesRequestHandler(_brokerConf, brokerId, _routingManager,
           _accessControlFactory, _queryQuotaManager, tableCache, queryDispatcher);
     }
+
+    LOGGER.info("Initializing PinotFSFactory");
+    PinotFSFactory.init(_brokerConf.subset(CommonConstants.Broker.PREFIX_OF_CONFIG_OF_PINOT_FS_FACTORY));
+
+    LOGGER.info("Initialize ResultStore");
+    PinotConfiguration resultStoreConfiguration =
+        _brokerConf.subset(CommonConstants.CursorConfigs.PREFIX_OF_CONFIG_OF_RESULT_STORE);
+    try {
+      ResponseSerde responseSerde = ResponseSerdeService.getInstance().getResponseSerde(
+          resultStoreConfiguration.getProperty(CommonConstants.CursorConfigs.RESULT_STORE_SERDE,
+              CommonConstants.CursorConfigs.DEFAULT_RESULT_SERDE));
+      responseSerde.init(resultStoreConfiguration.subset(CommonConstants.CursorConfigs.RESULT_STORE_SERDE)
+          .subset(responseSerde.getType()));
+
+      String expirationTime = getConfig().getProperty(CommonConstants.CursorConfigs.RESULTS_EXPIRATION_INTERVAL,
+          CommonConstants.CursorConfigs.DEFAULT_RESULTS_EXPIRATION_INTERVAL);
+
+      _resultStore = (AbstractResponseStore) ResponseStoreService.getInstance().getResultStore(
+          resultStoreConfiguration.getProperty(CommonConstants.CursorConfigs.RESULT_STORE_TYPE,
+              CommonConstants.CursorConfigs.DEFAULT_RESULT_STORE_TYPE));
+      _resultStore.init(resultStoreConfiguration.subset(_resultStore.getType()), _brokerMetrics, responseSerde);
+    } catch (Exception e) {
+      LOGGER.error("Exception when create Cursor ResultStore. Creating default result store. {}", e.getMessage());
+      // TODO: Exit ?
```

Comment:
Probably we should exit, yes.

Reply:
I mean, not callyng System.exit, but shutting down the broker properly.

Reply:
I've removed the try-catch. So the exception wont be ignored.